src/generators/typescript/TypeScriptGenerator.ts

```diff
@@ -39,15 +41,29 @@ export class TypeScriptGenerator extends AbstractGenerator<TypeScriptOptions,Typ
   ) {
     super('TypeScript', TypeScriptGenerator.defaultOptions, options);
   }
-  
+
   /**
    * Render a complete model result where the model code, library and model dependencies are all bundled appropriately.
    *
    * @param model
    * @param inputModel
    * @param options
    */
-  async renderCompleteModel(model: CommonModel, inputModel: CommonInputModel, options: TypeScriptRenderCompleteModelOptions): Promise<RenderOutput> {
+  async renderCompleteModel(model: CommonModel, inputModel: CommonInputModel, {moduleSystem = 'ESM', exportType = 'default'}: TypeScriptRenderCompleteModelOptions): Promise<RenderOutput> {
+    // Shallow copy presets so that we can restore it once we are done
+    const originalPresets = [...(this.options.presets ? this.options.presets : [])];
+
+    // Add preset that adds the `export` keyword if needed
+    if (
+      moduleSystem === 'ESM' &&
+      exportType === 'named' &&
+      (originalPresets[0] !== TS_EXPORT_KEYWORD_PRESET ||
+        (Object.prototype.hasOwnProperty.call(originalPresets[0], 'preset') &&
+          (originalPresets[0] as PresetWithOptions).preset !== TS_EXPORT_KEYWORD_PRESET))
+    ) {
```

Comment:
@gabormagyar do you mind moving this to a helper function such as you suggested `hasPreset`, that way we don't clutter up this function, and in the future, we probably need this in other generators 🙂